gtdynamics/universal_robot/Joint.cpp

```diff
@@ -326,4 +345,50 @@ gtsam::Expression<gtsam::Vector6> Joint::poseConstraint(
   return gtdynamics::logmap(wTc, wTc_hat);
 }
 
-}  // namespace gtdynamics
\ No newline at end of file
+/* ************************************************************************* */
+gtsam::Vector6_ Joint::twistConstraint(uint64_t t) const {
+  gtsam::Vector6_ twist_p(internal::TwistKey(parent()->id(), t));
+  gtsam::Vector6_ twist_c(internal::TwistKey(child()->id(), t));
+  gtsam::Double_ q(internal::JointAngleKey(id(), t));
+  gtsam::Double_ qVel(internal::JointVelKey(id(), t));
+
+  gtsam::Vector6_ twist_c_hat(
+      std::bind(&Joint::transformTwistTo, this, child(), std::placeholders::_1,
+                std::placeholders::_2, std::placeholders::_3,
+                std::placeholders::_4, std::placeholders::_5,
+                std::placeholders::_6),
+      q, qVel, twist_p);
+
+  return twist_c_hat - twist_c;
+}
+
+/* ************************************************************************* */
+gtsam::Vector6_ Joint::wrenchEquivalenceConstraint(uint64_t t) const {
+  gtsam::Vector6_ wrench_p(internal::WrenchKey(parent()->id(), id(), t));
+  gtsam::Vector6_ wrench_c(internal::WrenchKey(child()->id(), id(), t));
+  gtsam::Double_ q(internal::JointAngleKey(id(), t));
+
+  gtsam::Vector6_ wrench_c_hat(
+      std::bind(&Joint::transformWrenchCoordinate, this, child(), std::placeholders::_1,
+                std::placeholders::_2, std::placeholders::_3,
```

Comment:
Think about simplifying with a function that takes expressions, a la "logmap" we added.
At the very least, say "using std::placeholders" above.

Reply:
There's no logmap associated with this function, as wrenches are Vector6s quantities, so just subtraction shall work.

I tried with using std::placeholders before, it compiles on mac, but does not work on ubuntu, as it reports that boost also has placeholders, and it got confused about which one to use.